contrib/babelfishpg_tsql/src/pl_exec-2.c

```diff
@@ -877,15 +878,32 @@ exec_stmt_exec(PLtsql_execstate *estate, PLtsql_stmt_exec *stmt)
 			i = 0;
 			foreach(lc, funcargs)
 			{
-				Node	   *n = lfirst(lc);
+				Node 	*n = lfirst(lc);
+				bool 	isFound = false;
 
 				if (argmodes &&
 					(argmodes[i] == PROARGMODE_INOUT ||
 					 argmodes[i] == PROARGMODE_OUT))
 				{
+					ListCell *paramcell;
+					j = 0;
+
+					foreach(paramcell, stmt->params)
+					{
+						tsql_exec_param *p = (tsql_exec_param *) lfirst(paramcell);
+						if (argnames[i] && p->name && strcmp(argnames[i], p->name) == 0)
+						{
+							isFound = true;
```

Comment:
Do we really need another variable for this?
Why can't you initialise j to -1 and then if you match this condition then set j.
Use j conditionally later